plugins/woocommerce/src/Internal/Admin/Settings/PaymentProviders/PayPal.php

```diff
@@ -0,0 +1,44 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\Admin\Settings\PaymentProviders;
+
+use WC_Payment_Gateway;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * PayPal payment gateway provider class.
+ *
+ * This class handles all the custom logic for the PayPal payment gateway provider.
+ */
+class PayPal extends PaymentGateway {
+
+	/**
+	 * Try to determine if the payment gateway is in test mode.
+	 *
+	 * This is a best-effort attempt, as there is no standard way to determine this.
+	 * Trust the true value, but don't consider a false value as definitive.
+	 *
+	 * @param WC_Payment_Gateway $payment_gateway The payment gateway object.
+	 *
+	 * @return bool True if the payment gateway is in test mode, false otherwise.
+	 */
+	public function is_in_test_mode( WC_Payment_Gateway $payment_gateway ): bool {
```

Comment:
great! should we check if it works? 🙂

Reply:
Yes. Install the extension and use a sandbox account.

Reply:
I've tried. It's broken for automattic.com emails
<img width="474" alt="image" src="https://github.com/user-attachments/assets/775a3a48-d309-4f9a-9145-6df56749fa1a" />

Reply:
I don't know if we have sandboxes for us. It is a bit convoluted to setup a sandbox dev account.